sigstore/verify/verifier.py

```diff
@@ -495,15 +553,58 @@ def verify_artifact(
         #      the other bundle materials (and input being verified).
         entry = bundle.log_entry
 
-        expected_body = _hashedrekord_from_parts(
-            bundle.signing_certificate,
-            bundle._inner.message_signature.signature,  # type: ignore[union-attr]
-            hashed_input,
-        )
-        actual_body = rekor_types.Hashedrekord.model_validate_json(
-            base64.b64decode(entry.body)
-        )
-        if expected_body != actual_body:
-            raise VerificationError(
-                "transparency log entry is inconsistent with other materials"
+        if (
+            entry._kind_version.kind == "hashedrekord"
+            and entry._kind_version.version == "0.0.2"
+        ):
+            # actual_body = v2.Entry().from_json(base64.b64decode(entry.body))
+            # A potential issue with the returned body not parsing properly.
+            # perhaps the json_name in the protos.
+            _dict = json.loads(base64.b64decode(entry.body))
+            actual_body = v2.Entry(
+                kind=_dict["kind"],
+                api_version=_dict["apiVersion"],
+                spec=v2.Spec(
+                    hashed_rekord_v0_0_2=v2.HashedRekordLogEntryV002.from_dict(
+                        _dict["spec"]["hashedRekordV002"]
+                    )
+                ),
+            )
+            expected_body = v2.Entry(
+                kind=entry._kind_version.kind,
+                api_version=entry._kind_version.version,
+                spec=v2.Spec(
+                    hashed_rekord_v0_0_2=v2.HashedRekordLogEntryV002(
+                        data=v1.HashOutput(
+                            algorithm=bundle._inner.message_signature.message_digest.algorithm,
+                            digest=bundle._inner.message_signature.message_digest.digest,
+                        ),
+                        signature=v2.Signature(
+                            content=bundle._inner.message_signature.signature,
+                            verifier=v2.Verifier(
+                                public_key=v2.PublicKey(
+                                    raw_bytes=bundle.signing_certificate.public_key().public_bytes(
+                                        encoding=serialization.Encoding.DER,
+                                        format=serialization.PublicFormat.SubjectPublicKeyInfo,
```

Comment:
this does not look right: if the bundle contains a signing certificate, we should expect the signature to contain that (and not just the public key of the certificate), right?

Reply:
I've used the certificate in my PR, I think it looks ok